### PR TITLE
opensora-pku 1.1 add sequence parallelism

### PR DESCRIPTION
**Pull request by zhanghuiyao**
# What does this PR do?

Fixes # (issue)

Adds # (feature)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [ ] Did you read the [contributor guideline](https://github.com/mindspore-lab/mindone/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the documentation with your changes? E.g. record bug fixes or new features in `What's New`. Here are the
      [documentation guidelines](https://github.com/mindspore-lab/mindcv/wiki/%E6%96%87%E6%A1%A3%E7%BC%96%E5%86%99%E8%A7%84%E8%8C%83)
- [ ] Did you build and run the code without any errors?
- [ ] Did you report the running environment (NPU type/MS version) and performance in the doc? (better record it for data loading, model inference, or training tasks)
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@xxx

### CODE REVIEW

File: examples/opensora_pku/README.md [hunk 1]
```diff
@@ -313,7 +313,7 @@ Then please revise `scripts/text_condition/train_videoae_221x512x512.sh`, and ch
 
 ```bash
 # start 221x512x512 finetuning, 8 NPUs
-bash scripts/text_condition/train_videoae_221x512x512.sh
```
Comment (wtomin) [inline examples/opensora_pku/README.md:316]: Since 221x512x512 optimizer parallelism leads to oom, how about deleting `train_videoae_221x512x512.sh` and only keep `train_videoae_221x512x512_sp.sh` ?

File: examples/opensora_pku/opensora/acceleration/communications.py [hunk 2]
```diff
@@ -0,0 +1,266 @@
...
+
+
+class AllToAll_SBH(_SingleAll2ALL):
+    def __init__(self, scatter_dim: int = 1, gather_dim: int = 0):
+        super(AllToAll_SBH, self).__init__(scatter_dim=scatter_dim, gather_dim=gather_dim)
+
+
+def prepare_parallel_data(
+    hidden_states, noise, encoder_hidden_states, attention_mask, encoder_attention_mask, use_image_num
+):
```
Comment (wtomin) [inline examples/opensora_pku/opensora/acceleration/communications.py:125]: Can you give some docstrings for the function `prepare_parallel_data` ? It alters the inputs' shapes to the LatteT2V and introduces two variables `temp_attention_mask` and `loss_mask`. These are critical changes and some explanations are favorable.

File: examples/opensora_pku/opensora/models/diffusion/latte/modeling_latte.py [hunk 3]
```diff
@@ -454,6 +468,18 @@ def construct(
...
         timestep_temp = timestep.repeat_interleave(num_patches, dim=0)
 
+        # BS H -> S B H
+        if get_sequence_parallel_state():
+            timestep_temp = timestep_temp.view(input_batch_size * num_patches, 6, -1).swapaxes(0, 1).contiguous()
+            # f, h -> f, 1, h
+            temp_pos_embed = self.temp_pos_embed[self.temp_pos_st : self.temp_pos_ed].unsqueeze(1)
+        else:
+            temp_pos_embed = self.temp_pos_embed[: self.video_length].unsqueeze(0)
+            if temp_pos_embed.shape[1] != frame:
```
Comment (wtomin) [inline examples/opensora_pku/opensora/models/diffusion/latte/modeling_latte.py:478]: In which case that `frame` is greater than `temp_pos_embed.shape[1]`? If they are different, either the `temp_pos_embed` shape is wrong or the output shape of the vae encoder is wrong. I don't padding here is the correct way to solve this problem.

File: examples/opensora_pku/README.md (hunk 1 above)
Comment (zhanghuiyao) [reply examples/opensora_pku/README.md:316]: The main branch doesn't seem to provide script `train_videoae_221x512x512.sh` ?

File: examples/opensora_pku/opensora/models/diffusion/latte/modeling_latte.py (hunk 3 above)
Comment (zhanghuiyao) [reply examples/opensora_pku/opensora/models/diffusion/latte/modeling_latte.py:478]: This is the writing method that follows the official implementation, but perhaps an exception should be thrown directly here.
https://github.com/PKU-YuanGroup/Open-Sora-Plan/blob/hw/opensora/models/diffusion/latte/modeling_latte.py#L459

File: examples/opensora_pku/opensora/acceleration/communications.py [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+            + [
+                inp_shape[gather_dim] * sp_size,
+            ]
+            + inp_shape[gather_dim + 1 :]
+        )
+
+        return output
+
+
+class bak_SingleAll2ALL(nn.Cell):
```
[the remark this answers is not in the document]
Comment (zhanghuiyao) [reply examples/opensora_pku/opensora/acceleration/communications.py]: done

File: examples/opensora_pku/opensora/acceleration/communications.py (hunk 2 above)
Comment (zhanghuiyao) [reply examples/opensora_pku/opensora/acceleration/communications.py:63]: done

File: examples/opensora_pku/README.md (hunk 1 above)
Comment (wtomin) [reply examples/opensora_pku/README.md:316]: Yes, you are right.

File: examples/opensora_pku/opensora/acceleration/communications.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+    def construct(self, x):
+        return self.op(x)
+
+
+class AllToAll_SBH(_SingleAll2ALL):
+    def __init__(self, scatter_dim: int = 1, gather_dim: int = 0):
+        super(AllToAll_SBH, self).__init__(scatter_dim=scatter_dim, gather_dim=gather_dim)
+
+
+def prepare_parallel_data(
```
Comment (CaitinZhao) [inline examples/opensora_pku/opensora/acceleration/communications.py:61]: 这个是否考虑放dataset处理那里，vae的部分直接拿1/SP份数据，temporal的那部分也加SP，减下vae的计算量和显存

**Review by CaitinZhao — APPROVED** [review]

**Review by SamitHuang — APPROVED** [review]
